app/main.py

```diff
@@ -1,2 +1,50 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.length = 0
+        self.threshold = int(self.capacity * 2 / 3)
+        self.hashtable = [[] for _ in range(self.capacity)]
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.length == self.threshold:
+            self.resize_hashtable()
+        hashed_key = hash(key)
+        index = hashed_key % self.capacity
+        while True:
+            if not self.hashtable[index]:
```

Comment:
To improve readability add this line:
cell = self.hashtable[index]
And then inside checks use cell
```suggestion
            if not cell:
```

Reply:
The same in getitem